File: Python/programmers/etc/level2/점프와_순간_이동.py
def solution(n):
    cnt = 1
    while n != 1:
        if ~n&1:
            n >>= 1
        else:
            n -= 1
            cnt += 1
    return cnt


n = 5
print(solution(n)) # 2

n = 6
print(solution(n)) # 2

n = 5000
print(solution(n)) # 5

File: Java/programmers/etc/level2/점프와순간이동.java
package CodeTest.Java.programmers.etc.level2;

public class 점프와순간이동 {
    public int solution(int n) {
        int ans = 1;
        
        while (n != 1) {
            if (n%2 == 0) {
                n /= 2;
            } else {
                n--;
                ans++;
            }
        }
        
        return ans;
    }
}
